tests/test_std.rs

```diff
@@ -353,6 +356,37 @@ fn sorted_by() {
     it::assert_equal(v, vec![4, 3, 2, 1, 0]);
 }
 
+quickcheck! {
+    fn k_smallest_range(n: u64, m: u64, k: u64) -> () {
+        // Generate a random permutation of n..n+m
+        let i = {
+            let mut v: Vec<u64> = (n..n.saturating_add(m)).collect();
```

Comment:
I should probably be putting a reasonable bound on `m`, as we do allocate `8 * (m + min(m, k))` bytes in this test...